src/streaming/thumbnail/ThumbnailTracks.js

```diff
@@ -117,6 +165,60 @@ function ThumbnailTracks(config) {
                 }
             });
         }
+
+        if (useSegmentBase) {
+            segmentBaseLoader.loadSegments(representation, Constants.IMAGE, representation.indexRange, {}, function (segments, representation) {
+                var cache = [];
+                segments = normalizeSegments(segments, representation);
+                track.segmentDuration = segments[0].duration; //assume all segments have the same duration
+                track.readThumbnail = function (time, callback) {
+
+                    let cached = null;
+                    cache.some(el => {
+                        if (el.start <= time && el.end > time) {
+                            cached = el.url;
+                            return true;
+                        }
+                    });
+                    if (cached) {
+                        callback(cached);
+                    } else {
+                        segments.some((ss) => {
+                            if (ss.mediaStartTime <= time && ss.mediaStartTime + ss.duration > time) {
+                                const baseURL = baseURLController.resolve(representation.path);
+                                loader.load({
+                                    method: 'get',
+                                    url: baseURL.url,
+                                    request: {
+                                        range: ss.mediaRange,
+                                        responseType: 'arraybuffer'
+                                    },
+                                    onload: function (e) {
+                                        let info = boxParser.getSamplesInfo(e.target.response);
```

Comment:
Would it make sense implementing a basic cache mechanism at fragment level? Not concerned about the amount of requests done when moving the mouse over the controlbar (most of them are going to be served from the cache) but about the amount of fragments parsed. We could implement a basic cache mechanism that stores the result of the latest parsed fragment. Two or more consecutive requests of the same fragment (most probably thing) will require just one parsing.

What do you think?

Reply:
I don't know if that's completely necessary right now:
Currently every fragment only has only one sample (one tiled jpeg image). When we have an example stream using multiple samples per fragment, a better caching approach will definitely be necessary.
I will extend the implementation and create another pull request when have this ready.

Reply:
For reference, the container MP4 with the tiled thumbnails is now formatted as a `moov` box with a `udta` box which has the http://dashif.org/thumbnail_tile schemeIdUri, and then each individual tiling is is a `moof`/`mdat` fragment.

E.g. as xml output from `MP4Box -diso`:

```xml
[...]
  <MovieBox>
    <BoxInfo Size="665" Type="moov"/>
[...]
      <UserDataBox>
        <BoxInfo Size="57" Type="udta"/>
        <KindBox schemeURI="http://dashif.org/thumbnail_tile" value="3x4">
          <BoxInfo Size="49" Type="kind"/>
          <FullBoxInfo Version="0" Flags="0x0"/>
        </KindBox>
      </UserDataBox>
    </TrackBox>
  </MovieBox>
  <SegmentIndexBox reference_ID="1" timescale="12288" earliest_presentation_time="0" first_offset="0">
    <BoxInfo Size="188" Type="sidx"/>
    <FullBoxInfo Version="0" Flags="0x0"/>
    <Reference type="0" size="27241" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="21019" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="22880" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="29088" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="31269" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="27452" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="27317" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="27232" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="27726" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="29735" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="29772" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="33287" duration="737280" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
    <Reference type="0" size="12771" duration="172032" startsWithSAP="1" SAP_type="1" SAPDeltaTime="0"/>
  </SegmentIndexBox>
  <MovieFragmentBox TrackFragments="1">
    <BoxInfo Size="104" Type="moof"/>
[...]
  </MovieFragmentBox>
  <MediaDataBox dataSize="27129">
    <BoxInfo Size="27137" Type="mdat"/>
  </MediaDataBox>
  <MovieFragmentBox TrackFragments="1">
    <BoxInfo Size="104" Type="moof"/>
    <MovieFragmentHeaderBox FragmentSequenceNumber="2">
[...]
  </MovieFragmentBox>
  <MediaDataBox dataSize="20907">
    <BoxInfo Size="20915" Type="mdat"/>
  </MediaDataBox>
  <MovieFragmentBox TrackFragments="1">
    <BoxInfo Size="104" Type="moof"/>
    <MovieFragmentHeaderBox FragmentSequenceNumber="3">
[...]
```